addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -3173,6 +3147,42 @@ export class OdooEditor extends EventTarget {
         }
     }
 
+    /**
+     * Inserts a link in the editor. Called after pressing space or (shif +) enter.
+     * Performs a regex check to determine if the url has correct syntax.
+     */
+    _insertLink() {
+        const selection = this.document.getSelection();
+        if (
+            selection &&
+            selection.anchorNode &&
+            !closestElement(selection.anchorNode).closest('a') &&
+            selection.anchorNode.nodeType === Node.TEXT_NODE
+        ) {
+            const textSliced = selection.anchorNode.textContent.slice(0, selection.anchorOffset);
+            const textNodeSplitted = textSliced.split(/\s/);
+            let potentialUrl = textNodeSplitted.pop();
+            if (potentialUrl === '') {  // Ignore optional added space
+                potentialUrl = textNodeSplitted.pop();
```

Comment:
Previously, we always had a first `pop()` before retrieving the `potentialUrl` with a second `pop()`. This first pop had a comment "Remove added space", but that extra space is no longer present from the editor as we catch the event on `keydown` and not on `input`. However it is still needed for tests to run. See below.